### DISCUSSION TITLE
연동 회원에 대한 고민

### DISCUSSION THREAD

**Discussion by YurimYang** [SCBJ-7/SCBJ-BE#9] [Ideas]
### 안건

- 현재, 서비스 로직이 회원가입 -> 야놀자 연동 -> 예약내역 불러오기 로 진행된다.
- 회원 테이블과, 예약내역 테이블 사이에 회원연동 테이블을 만들어야할지 고민이 된다.
![image](https://github.com/SCBJ-7/SCBJ-BE/assets/63856521/c158a506-763d-4604-9597-4702faa9b238)
- 이유 : 야놀자 회원 데이터가 존재하지 않으므로 연동이 되었다는 전제하에 개발을 진행해야하기 때문이다.

### 고민사항

-  회원, 예약내역 사이에 "연동회원" 테이블을 생성해야할지 고민이 됩니다. 
1. 회원가입시, 회원 테이블에 데이터가 새롭게 INSERT 되지만 해당 회원의 예약내역이 있어야한다.
2. 그러므로, 연동하기 버튼을 클릭시, 연동회원 테이블에 회원ID, 예약내역ID 를 저장한다.
즉, 연동회원 테이블에는 회원ID, 예약내역ID를 FK로 갖는다. 
![image](https://github.com/SCBJ-7/SCBJ-BE/assets/63856521/9d904d0b-51a5-42bb-81b1-cfec22c58522)

**Comment by wocjf0513** [comment]
저는 다양한 방법이 있을 거 같은데 
1) 미리 야놀자 회원 테이블을 목 데이터로 채워놓고 (의도: 야놀자 회원이 있다고 가정) 
2) 예약 내역을 야놀자 회원 테이블과 관계를 맺게 한 다음 (의도: 야놀자 회원이 예약을 해서 관계를 맺고있다 가정)
3) 예약 내역에 목 데이터를 채운다. (의도: 야놀자 회원이 예약을 했다고 가정)

그리고 회원가입 -> 연동하기 -> 회원 가입으로 작성한 이메일이 기존 연동 회원 테이블에 정보가 있을 때만, 연동이 가능하게 하는 로직을 짜서
연동된 회원만, 예약 내역 데이터와 관계가 있으니 조회할 수 있게 하면 어떨까 싶습니다!!

**Reply by Kwonminwoo** [reply]
저도 이 방법이 좋을 것 같습니다. 연동 된다는 의미도 있고 간편하게 처리할 수 있을 것 같아요.
새로운 연동 회원을 테스트 할 때에는 야놀자 회원 테이블에 목 데이터를 추가해서 연동 되게 처리하면 될 것 같아요.
아니면 야놀자 연동과 관련된 기능 및 테이블을 최소화 하여 처리하는 것도 좋을 것 같습니다. 저희가 생각한 주 도메인에 포커싱 하여 고도화 하는게 더 좋을 것 같습니다.

**Reply by Aleexender** [reply]
저도 민우님 의견에 동의해요!
미리 야놀자 회원 테이블을 목으로 채워놓고
예약내역에 목 데이터를 연동되게 처리하면 좋을거같습니다!
지금 상황상 계획은 바꾸기엔 늦은거 같고 야놀자 테이블을 재철님 의견과 동일하게 최대한 간소화하고 간단하게 처리하고 한다면 딱히 변경같은게 일어나지도 않을거같아서 괜찮을거같아요!
